subsys/bluetooth/controller/hci_driver.c

```diff
@@ -244,38 +274,174 @@ void SIGNALLING_Handler(void)
+	clock_cfg.rc_temp_ctiv = BLE_CONTROLLER_RECOMMENDED_RC_TEMP_CTIV;
+
+	err = ble_controller_init(blectlr_assertion_handler,
+				  &clock_cfg,
+				  SWI5_IRQn);
+	return err;
+}
+
+static int ble_enable(void)
+{
+	int err;
+	ble_controller_cfg_t cfg;
+
+	cfg.master_count.count = 1;
+
+	err = ble_controller_cfg_set(BLE_CONTROLLER_DEFAULT_RESOURCE_CFG_TAG,
+				     BLE_CONTROLLER_CFG_TYPE_MASTER_COUNT,
+				     &cfg);
+	if (err < 0 || err > sizeof(ble_controller_mempool)) {
+		return err;
+	}
+
+	cfg.slave_count.count = 1;
+
+	err = ble_controller_cfg_set(BLE_CONTROLLER_DEFAULT_RESOURCE_CFG_TAG,
+				     BLE_CONTROLLER_CFG_TYPE_SLAVE_COUNT,
+				     &cfg);
+	if (err < 0 || err > sizeof(ble_controller_mempool)) {
+		return err;
+	}
+
+	cfg.buffer_cfg.rx_packet_size = 251;
+	cfg.buffer_cfg.tx_packet_size = 251;
+	cfg.buffer_cfg.rx_packet_count = 10;
+	cfg.buffer_cfg.tx_packet_count = 10;
+
+	err = ble_controller_cfg_set(BLE_CONTROLLER_DEFAULT_RESOURCE_CFG_TAG,
+				     BLE_CONTROLLER_CFG_TYPE_BUFFER_CFG,
+				     &cfg);
+	if (err < 0 || err > sizeof(ble_controller_mempool)) {
+		return err;
+	}
+
+	cfg.event_length.event_length_us = 7500;
+	err = ble_controller_cfg_set(BLE_CONTROLLER_DEFAULT_RESOURCE_CFG_TAG,
+				     BLE_CONTROLLER_CFG_TYPE_EVENT_LENGTH,
+				     &cfg);
+	if (err < 0 || err > sizeof(ble_controller_mempool)) {
+		return err;
+	}
+
+	err = MULTITHREADING_LOCK_ACQUIRE();
+	if (!err) {
+		err = ble_controller_enable(host_signal,
+					    ble_controller_mempool);
+		MULTITHREADING_LOCK_RELEASE();
+	}
+	if (err < 0) {
+		return err;
+	}
+
+	/* Start processing software interrupts. This enables, e.g., the flash
+	 * API to work without having to call bt_enable(), which in turn calls
+	 * hci_driver_open().
+	 *
+	 * FIXME: Here we possibly start dynamic behavior during initialization,
+	 * which in general is a bad thing.
+	 */
+	k_thread_create(&signal_thread_data, signal_thread_stack,
+			K_THREAD_STACK_SIZEOF(signal_thread_stack),
+			signal_thread, NULL, NULL, NULL,
+			K_PRIO_COOP(CONFIG_BLECTLR_PRIO), 0, K_NO_WAIT);
+
+	return 0;
+}
+
 static int hci_driver_init(struct device *unused)
 {
 	ARG_UNUSED(unused);
 
-	u32_t err = blectlr_init(host_signal);
+	bt_hci_driver_register(&drv);
 
-	if (err) {
-		/* Probably memory */
-		return -ENOMEM;
+	s32_t err = 0;
+
+	err = ble_enable();
+
+	if (err < 0) {
+		return err;
 	}
 
-	bt_hci_driver_register(&drv);
+	IRQ_DIRECT_CONNECT(RADIO_IRQn, BLE_CONTROLLER_IRQ_PRIO_HIGH,
+			   ble_controller_RADIO_IRQHandler, IRQ_ZERO_LATENCY);
+	IRQ_DIRECT_CONNECT(RTC0_IRQn, BLE_CONTROLLER_IRQ_PRIO_HIGH,
+			   ble_controller_RTC0_IRQHandler, IRQ_ZERO_LATENCY);
+	IRQ_DIRECT_CONNECT(TIMER0_IRQn, BLE_CONTROLLER_IRQ_PRIO_HIGH,
+			   ble_controller_TIMER0_IRQHandler, IRQ_ZERO_LATENCY);
+	IRQ_DIRECT_CONNECT(POWER_CLOCK_IRQn, BLE_CONTROLLER_IRQ_PRIO_LOW,
+			   ble_controller_POWER_CLOCK_IRQHandler,
+			   IRQ_ZERO_LATENCY);
+	IRQ_CONNECT(SWI5_IRQn, BLE_CONTROLLER_IRQ_PRIO_LOW,
```

Comment:
Ideally the SWI IRQ number should configurable.

Reply:
SWI is not currently in Device Tree. And other drivers use hardcoded numbers.
Let's leave this for now and add SWI to device tree and update users correctly later instead of doing something wierd here.